test/org/sagebionetworks/bridge/validators/CompoundActivityDefinitionValidatorTest.java

```diff
@@ -0,0 +1,163 @@
+import org.sagebionetworks.bridge.dynamodb.DynamoCompoundActivityDefinition;
+import org.sagebionetworks.bridge.exceptions.InvalidEntityException;
+import org.sagebionetworks.bridge.models.schedules.CompoundActivityDefinition;
+import org.sagebionetworks.bridge.models.schedules.SchemaReference;
+import org.sagebionetworks.bridge.models.schedules.SurveyReference;
+
+public class CompoundActivityDefinitionValidatorTest {
+    private static final List<SchemaReference> SCHEMA_LIST = ImmutableList.of(new SchemaReference("test-schema",
+            null));
+    private static final List<SurveyReference> SURVEY_LIST = ImmutableList.of(new SurveyReference("test-survey",
+            "test-survey-guid", null));
+    private static final String TASK_ID = "test-task";
+    private static final CompoundActivityDefinitionValidator VALIDATOR = new CompoundActivityDefinitionValidator(
+            ImmutableSet.of(TASK_ID));
+
+    // branch coverage
+    @Test
+    public void validatorSupportsClass() {
+        assertTrue(VALIDATOR.supports(CompoundActivityDefinition.class));
+    }
+
+    // branch coverage
+    @Test
+    public void validatorSupportsSubclass() {
+        assertTrue(VALIDATOR.supports(DynamoCompoundActivityDefinition.class));
+    }
+
+    // branch coverage
+    @Test
+    public void validatorDoesntSupportClass() {
+        assertFalse(VALIDATOR.supports(String.class));
+    }
+
+    // branch coverage
+    // we call the validator directly, since Validate.validateThrowingException filters out nulls and wrong types
+    @Test
+    public void validateNull() {
+        MapBindingResult errors = new MapBindingResult(new HashMap<>(), "CompoundActivityDefinition");
+        VALIDATOR.validate(null, errors);
+        assertTrue(errors.hasErrors());
+    }
+
+    // branch coverage
+    // we call the validator directly, since Validate.validateThrowingException filters out nulls and wrong types
+    @Test
+    public void validateWrongClass() {
+        MapBindingResult errors = new MapBindingResult(new HashMap<>(), "CompoundActivityDefinition");
+        VALIDATOR.validate("wrong class", errors);
+        assertTrue(errors.hasErrors());
+    }
+
+    @Test
+    public void valid() {
+        Validate.entityThrowingException(VALIDATOR, makeValidDef());
+    }
+
+    @Test
+    public void nullStudyId() {
+        blankStudyId(null);
+    }
+
+    @Test
+    public void emptyStudyId() {
+        blankStudyId("");
+    }
+
+    @Test
+    public void blankStudyId() {
+        blankStudyId("   ");
+    }
+
+    private static void blankStudyId(String studyId) {
+        CompoundActivityDefinition def = makeValidDef();
+        def.setStudyId(studyId);
+
+        try {
+            Validate.entityThrowingException(VALIDATOR, def);
+            fail("expected exception");
+        } catch (InvalidEntityException ex) {
+            assertTrue(ex.getMessage().contains("studyId must be specified"));
+        }
+    }
```

Comment:
FWIW I created a TestUtils.assertValidatorMessage(Validator validator, Object object, String fieldName, String error) method to use on these kinds of tests